src/poseidon_gencontract.js

```diff
@@ -117,29 +116,29 @@ export function createCode(nInputs) {
 
     saveM();
 
-    C.push("0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001");  // q
+    C.push("0x0800000000000011000000000000000000000000000000000000000000000001");  // q this is the big prime in starknet
```

Comment:
so the maximal prime number supported on starknet changed recently? or maybe the previous one didn't correspond to starknet

Reply:
right polygons Poseidon is over a different field than Starkwares among other changes. Poseidon comes in different flavors, there is no single unifying implementation but they do share a core structure I guess(am not an expert per say). Every Poseidon implementation varies with MDS matrix, rounds constants and the box function.